src/dbnode/persist/fs/commitlog/reader.go

```diff
@@ -207,9 +205,64 @@ func (r *reader) readLogEntry() error {
 	return nil
 }
 
+func (r *reader) namespaceIDReused(id []byte) ident.ID {
+	var namespaceID ident.ID
+	for _, ns := range r.namespacesRead {
+		if bytes.Equal(ns.namespaceIDBytes, id) {
+			namespaceID = ns.namespaceIDRef
+			break
+		}
+	}
+	if namespaceID == nil {
+		// Take a copy and keep around to reuse later.
+		namespaceBytes := append([]byte(nil), id...)
```

Comment:
it's meant to be just as fast (although let me preallocate here, since I'm not): https://gist.github.com/xogeny/b819af6a0cf8ba1caaef#gistcomment-2822939

i.e. I'll change this to `append(make([]byte, len(id)), id...)`